src/elements/date-picker/DatePicker.js

```diff
@@ -131,22 +142,71 @@ export class DatePicker {
 
     select(event, day, fireEvents) {
         swallowEvent(event);
-
-        this.selected = day.date;
-        this.updateHeading();
-
-        if (fireEvents) {
+        if (this.range) {
+            if (!this.selected && event) {
+                this.selected = day.date;
+            } else if (this.selected && this.selected2) {
+                this.selected = day.date;
```

Comment:
The problem is selected is used for the normal calendar as well. I could change selected2 to selectedEnd if that makes more sense for you? I am open to ideas.